### PR TITLE
Feature/g UI ld 1141 replace build yml hack

### PR DESCRIPTION
**Pull request by Jorg88**

### CODE REVIEW

File: src/mpyl/steps/steps.py [hunk 1]
```diff
@@ -112,9 +112,16 @@ def _execute(
...
+        project: Project,
+        stages: list[Stage],
+        required_artifact: Optional[ArtifactType],
+        deploy_image: Optional[str] = None,
     ) -> Optional[Artifact]:
-        if not required_artifact or required_artifact == ArtifactType.NONE:
+        if (
+            not required_artifact
+            or required_artifact == ArtifactType.NONE
+            or (required_artifact == ArtifactType.DOCKER_IMAGE and deploy_image)
```
Comment (ptab) [inline src/mpyl/steps/steps.py:123]: Am I reading this wrong, or should this be 
```suggestion
            or (required_artifact == ArtifactType.DOCKER_IMAGE and not deploy_image)
```
?

Comment (ptab) [reply src/mpyl/steps/steps.py:123]: Ah no, ignore the previous comment.

Comment (ptab) [reply src/mpyl/steps/steps.py:123]: ([This method is planned to go away soon anwyay](https://github.com/Vandebron/gh-mpyl/pull/77) 😄)

**Review by ptab — APPROVED** [review]

File: src/mpyl/steps/steps.py (hunk 1 above)
Comment (Jorg88) [reply src/mpyl/steps/steps.py:123]: What a rollercoaster of comments! Haha

**Pull request merged**